lsp-mode.el

```diff
@@ -4217,6 +4228,20 @@ Also, additional data to attached to each candidate can be passed via PLIST."
     (gethash "documentation")
     (lsp--render-element)))
 
+(defun lsp--capf-get-context (trigger-characters)
+  "Get completion context with provided TRIGGER-CHARACTERS."
+  (let* (trigger-char
+         (trigger-kind (cond
+                        ((setq trigger-char (lsp--looking-back-trigger-characters-p
+                                             trigger-characters))
+                         'character)
+                        ((equal lsp--capf-cache 'incomplete) 'incomplete)
+                        (t 'invoked))))
+    (ht<-alist
```

Comment:
do we need this conversion?

Reply:
Yes, the idea is to only include `triggerCharacter` when it existed. Some server like `clangd` basically just return nil if we pass `triggerCharacter` with nil

Reply:
I don't understand how converting alist to ht will help for that?

Reply:
The `append` will not add the `nil` element to `alist`. Then we we can convert that list to `ht` with only non-nil member.
Is there a better way to write that? I don't think converting from a list to ht take more time, especially since these operations are not run that frequently

Reply:
I have approved the PR as it is. 

> Then we we can convert that list to ht with only non-nil member.

Append part is fine - my question is solely about the alist->ht conversion. AFAICS this conversion does not strip nils. 

(But in general, feel free to merge it as it is).

Reply:
Thanks, the main part of this is `append` to not include nil. then `ht<-alist` just goes ahead and do the conversion like what `ht` does.